tests/test_postcodes.py

```diff
@@ -58,6 +58,19 @@ def test_extract_postcodes_list_of_matches():
     assert postcodes == ["BN9 0DF", "BN9 9BP", "BN9 9BN", "BN9 9QD", "BN9 0AS", "BN9 9PA", "BN9 9PD", "BN9 0DF"]
 
 
+def test_extract_postcodes_list_of_matches_2():
```

Comment:
Can we give this test a more descriptive name? i.e. describe how it is different from the test above?

Reply:
Changed the name of the test and added a description below.

Reply:
@samuelhwilliams 
Sam, for some reason I can't reply to your message, that's why i'm writing here.
As I see it, both the extract_postcodes and postcode_list uses the same regex. Yes, it would be better to use ^ and $ anchors to check for the beginning and the end of a postcode, but it will not work with the extract_postcodes function.
In cases when we have short strings which should contain only one single postcode, like in the case of postcode_list, and we check these one-by-one, I think it would be the best and the easiest check to use. 

But in case of the extract_postcodes function it would not work. This function can have long strings as argument and it has to find and return all postcodes from this string. 
In tests/test_postcodes.py/test_extract_postcodes_list_of_matches there's a long string with several postcodes, each starting in a new line. In this case we can't use ^ and & to check for boundaries, that's why I used /b checking for word boundaries and added the part to the regex to allow for the new line characters. 

If we want to use the same regex for both functions, I don't know how we could use tightly-bound checking in it.
Maybe having two separate regular expressions, one with ^ and &, and the other with word boundary checking /b.
Then in postcode_list to use the tightly-bound expression directly, and in extract_postcodes first extract postcodes with the word boundary checking first, and then check again with the tightly-bound regex, but in this case it's a double work to do.